mountpoint-s3/src/inode.rs

```diff
@@ -1239,7 +1243,7 @@ impl Inode {
     fn get_inode_state(&self) -> Result<RwLockReadGuard<InodeState>, InodeError> {
         let inode_state = self.inner.sync.read().unwrap();
         match &inode_state.kind_data {
-            InodeKindData::Directory { deleted, .. } if *deleted => Err(InodeError::InodeDoesNotExist(self.ino())),
+            InodeKindData::Directory { deleted, .. } if *deleted => Err(InodeError::DirectoryDoesNotExist(self.err())),
```

Comment:
I'm not too convinced about this change. We would return a different error depending on whether the inode was actually removed or just flagged with `deleted`. So far, this has been an implementation detail.

Reply:
I thought to do is to cover as much case as possible to log filename. But you are right, deleted is only supposed to be implementation detail.